pkg/install/hypershift.go

```diff
@@ -146,20 +148,30 @@ func (c *UpgradeController) RunHypershiftCleanup(ctx context.Context) error {
 }
 
 // This is run when:
-//   1) the controller starts up
-//   2) when the hypershift installation options (secrets and configmap) change
-func (c *UpgradeController) RunHypershiftInstall(ctx context.Context) error {
-	c.log.Info("enter runHypershiftInstall")
+//   1) the controller starts up, controllerStartup = true
+//   2) when the hypershift installation options (secrets and configmap) change, controllerStartup = false
+func (c *UpgradeController) RunHypershiftInstall(ctx context.Context, controllerStartup bool) error {
+	c.log.Info("enter runHypershiftInstall, controllerStartup = " + strconv.FormatBool(controllerStartup))
 	defer c.log.Info("exit runHypershiftInstall")
 
-	if err, ok := c.deploymentUpgradable(ctx); !ok || err != nil {
+	err, ok, operatorDeployment := c.operatorUpgradable(ctx)
+
+	if !ok || err != nil {
```

Comment:
nit picking, the following is easier to read
```go
if err != nil {
  return err
} else if !ok {
  c.log.Info("hypershift operator exists but not deployed by addon, skip update")
  return nil
}
```

Reply:
Good one!